crates/comms/src/livekit_web.rs

```diff
@@ -0,0 +1,279 @@
+
+    #[wasm_bindgen(catch)]
+    fn send_audio_frame(
+        track: &JsValue,
+        samples: &[f32],
+        sample_rate: u32,
+        num_channels: u32,
+    ) -> Result<(), JsValue>;
+
+    #[wasm_bindgen(catch)]
+    fn set_room_event_handler(
+        room: &JsValue,
+        handler: &Closure<dyn FnMut(JsValue)>,
+    ) -> Result<(), JsValue>;
+
+    #[wasm_bindgen(catch)]
+    fn set_microphone_enabled(enabled: bool) -> Result<(), JsValue>;
+
+    #[wasm_bindgen(catch)]
+    fn is_microphone_available() -> Result<bool, JsValue>;
+}
+
+pub struct LivekitWebPlugin;
+
+impl Plugin for LivekitWebPlugin {
+    fn build(&self, app: &mut App) {
+        app.add_systems(Update, update_mic_state);
+    }
+}
+
+fn update_mic_state(
+    mut mic_state: ResMut<MicState>,
+    mut last_enabled: Local<Option<bool>>,
+    mut last_available: Local<Option<bool>>,
+) {
+    // Check if microphone is available in the browser
+    let current_available = is_microphone_available().unwrap_or(false);
+
+    // Only update availability if it changed
+    if last_available.is_none() || last_available.unwrap() != current_available {
+        mic_state.available = current_available;
+        *last_available = Some(current_available);
+    }
+
+    // Only update microphone enabled state if it changed
+    if last_enabled.is_none() || last_enabled.unwrap() != mic_state.enabled {
+        if let Err(e) = set_microphone_enabled(mic_state.enabled) {
+            warn!("Failed to set microphone state: {:?}", e);
+        }
+        *last_enabled = Some(mic_state.enabled);
+    }
+}
+
+pub fn livekit_handler_inner(
+    transport_id: Entity,
+    remote_address: &str,
+    app_rx: Arc<Mutex<Receiver<NetworkMessage>>>,
+    sender: Sender<PlayerUpdate>,
+) -> Result<(), anyhow::Error> {
+    debug!(">> lk connect async : {}", remote_address);
+
+    let url = Uri::try_from(remote_address).unwrap();
+    let address = format!(
+        "{}://{}{}",
+        url.scheme_str().unwrap_or_default(),
+        url.host().unwrap_or_default(),
+        url.path()
+    );
+    let params = HashMap::from_iter(url.query().unwrap_or_default().split('&').flat_map(|par| {
+        par.split_once('=')
+            .map(|(a, b)| (a.to_owned(), b.to_owned()))
+    }));
+    debug!("{:?}", params);
+    let token = params.get("access_token").cloned().unwrap_or_default();
+
+    // In WASM, we can't block or create threads, so we just spawn the async task
+    spawn_local(async move {
+        if let Err(e) = run_livekit_session(transport_id, &address, &token, app_rx, sender).await {
+            error!("LiveKit session error: {:?}", e);
+        }
+    });
+
+    Ok(())
+}
+
+async fn run_livekit_session(
+    transport_id: Entity,
+    address: &str,
+    token: &str,
+    app_rx: Arc<Mutex<Receiver<NetworkMessage>>>,
+    sender: Sender<PlayerUpdate>,
+) -> Result<(), anyhow::Error> {
+    let room = connect_room(address, token)
```

Comment:
can we please wrap this in a retry loop, similar to native in `livekit_handler`. the native version repeatedly connects until the sender is closed (i.e when we change island / scene).